lib/system_description_validator.rb

```diff
@@ -16,195 +16,75 @@
 # you may find current contact information at www.suse.com
 
 class SystemDescriptionValidator
-  def initialize(description)
-    @description = description
-  end
-
-  def validate_json(json)
-    format_version = json["meta"]["format_version"]
+  def initialize(hash, path = nil)
```

Comment:
The responsibilities are a bit vague here. If the `SystemDescriptionValidator` has the responsibility to validate a `SystemDescription` it should take a `SystemDescription` object as parameter, not a hash and a path.

If the responsibilities of validating the manifest and the files are so separate that they are not interacting with each other, maybe it would be better to merge the code into the `JsonValidator` and the `FileValidator` and call them separately. If this is happening from within the `SystemDescription` it could be the preferable solution. If it is happening outside, it could be better to not expose the details of how a system description is internally represented, and work on the `SystemDescription` object.